controllers/activemqartemis_reconciler.go

```diff
@@ -1764,17 +1764,29 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewPodTemplateSpecForCR(customR
 
 	clog.V(1).Info("Total volumes ", "volumes", podSpec.Volumes)
 
-	// this depends on init container passing --java-opts to artemis create via launch.sh *and* it
-	// not getting munged on the way. We CreateOrAppend to any value from spec.Env
 	var mountPoint = secretPathBase
 	if !isSecret {
 		mountPoint = cfgMapPathBase
 	}
-	javaOpts := corev1.EnvVar{
-		Name:  "JAVA_OPTS",
-		Value: brokerPropertiesConfigSystemPropValue(mountPoint, brokerPropertiesResourceName, brokerPropertiesMapData),
+	brokerPropsValue := brokerPropertiesConfigSystemPropValue(mountPoint, brokerPropertiesResourceName, brokerPropertiesMapData)
+
+	// only use init container JAVA_OPTS on existing deployments and migrate to JDK_JAVA_OPTIONS for independence
+	// from init containers and broker run scripts
+	if environments.Retrieve(podSpec.InitContainers, "JAVA_OPTS") != nil {
```

Comment:
is this check required to avoid broker pod restarts after the operator upgrade?

Reply:
exactly. if there is any use of JAVA_OPTS, presumably they work and lets piggy back on that and not force a restart by a env change to the SS. 
If not, this is a new CR so we can go with the new approach.